datafusion/expr/src/signature.rs

```diff
@@ -46,6 +46,8 @@ pub enum TypeSignature {
     // A function such as `array` is `VariadicEqual`
     // The first argument decides the type used for coercion
     VariadicEqual,
+    /// arbitrary number of arguments of an arbitrary type
```

Comment:
This means to say all arguments can have different types right? I suggest:
```suggestion
    /// arbitrary number of arguments with arbitrary types
```